### DISCUSSION TITLE
在@Listener监听的方法中，怎么阻止后续优先级低的监听方法执行呢

### DISCUSSION THREAD

**Discussion by Starxkif** [simple-robot/Discussions#62] [Q&A] [answered]
比如我现在有很多个监听方法，但是它们都有一些共同的逻辑判断，我想把这些逻辑判断都提取到优先级高的监听方法中，只要高优先级的拒绝了，后续监听器便都不会执行了φ(゜▽゜*)♪

**Comment by ForteScarlet** [comment, marked as answer]
参考 [此文档](https://simbot.forte.love/docs/basic/event-listener#%E4%BA%8B%E4%BB%B6%E5%A4%84%E7%90%86%E7%BB%93%E6%9E%9C)，当监听函数的返回结果为 `EventResult` 类型且其属性 `isTruncated` 指为 `true` 时将会截断后续其他监听函数的执行。

例如可以：
```kotlin
// Kotlin
return EventResult.truncate()
// 或
return EventResult.of(isTruncated = true)
```

```java
// Java
return EventResult.truncate();
// 或
return EventResult.of(null, true);
```
可前往 [API文档](https://simple-robot-library.github.io/simbot3-main-apiDoc) 搜索 `EventResult` 查询API

**Reply by Starxkif** [reply]
还有一个疑问，监听函数之间可以传递消息吗，比如优先级高的监听函数处理过的消息，后续监听器可以获取到进行二次处理

**Reply by ForteScarlet** [reply]
参考文档 https://simbot.forte.love/docs/basic/event-listener#%E4%BC%A0%E9%80%92%E4%B8%8E%E8%8E%B7%E5%8F%96
